pkg/controller/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -108,6 +108,21 @@ func (r *ReconcileMasterUserRecord) Reconcile(request reconcile.Request) (reconc
 		logger.Error(err, "unable to get MasterUserRecord")
 		return reconcile.Result{}, err
 	}
+
+	// TODO remove this code block once all MUR resources have had their labels updated
+	migrated, err := r.migrateLabels(mur)
+	if err != nil {
+		// I think it's overkill to define a new status reason here?  At least log the error for now
+		logger.Error(err, "unable to migrate deprecated MUR labels")
+		return reconcile.Result{}, err
+	}
+
+	if migrated {
+		return reconcile.Result{
+			Requeue: true,
```

Comment:
I don't think we have to explicitly request a requeue. We watch MUR anyway: https://github.com/codeready-toolchain/host-operator/blob/master/pkg/controller/usersignup/usersignup_controller.go#L81-L88
And UserSignup reconcile will be called again as soon as MUR is updated.

Reply:
This case a bit different - this inside of `masteruserrecord_controller`, not in `usersignup_controller`, so if you would want to rely on usersignup controller functionality, then it should be moved there.
The `masteruserrecord_controller` watches changes in MURs only when the generation number is changed which is not the case for changes in the labels.

Reply:
To be honest, the cleaner solution for this would be moving the migration stuff to usersignup functionality - as I mentioned before: https://github.com/codeready-toolchain/host-operator/blob/master/pkg/controller/usersignup/masteruserrecord.go#L9

But since we are "only" changing the label, then it's not a big deal. On the other hand, this requeue is a bit confusing here, since we don't requeue when we change something in the `MUR.status` either - after all, it has more or less the same impact on the definition of the MUR resource.

Reply:
Ah right! I missed the fact that this migration code in the MUR controller. Not in the UserSignup one. And yeah.. I think it should be in the UserSignup controller.

Reply:
I can move the migration over to the UserSignup controller, however it would require a specific list operation on the MUR to find any of the older labels, an operation that would have to be run every single time we reconcile, whereas in the MUR controller the resource is given to you.